### ISSUE TITLE
Add TypeScript file definition

### ISSUE THREAD

**Issue by mt89vein** [autoNumeric/vue-autoNumeric#22]
Hi guys!

Can someone add typescript file definition ? :) It will be helpful for typescript users :)
something like that:
```ts
	class VueAutonumeric extends Vue {
		value: number | null | string
		options: object | string | Array<any>
		resetOnOptions: boolean
		placeholder: string
		tag: string
	}
```

thanks !)

**Comment by AlexandreBonneau** [comment]
@mt89vein Could you create a PR with the `.d.ts` file?

**Comment by AlexandreBonneau** [comment]
Hey @mt89vein, would you be willing to guide me in what needs to be done to please the typescript users? ;)
*ie. a PR would be great!*

**Comment by agentschmitt** [comment]
Without typescript definitions we will get an error when importing your library.
import VueAutoNumeric from 'vue-autonumeric';

Could not find a declaration file for module 'vue-autonumeric'. '....../node_modules/vue-autonumeric/dist/vue-autonumeric.min.js' implicitly has an 'any' type.
  Try `npm install @types/vue-autonumeric` if it exists or add a new declaration (.d.ts) file containing `declare module 'vue-autonumeric';`

To fix this your package should reference a definition file to describe your component.
I will do an pull request for this.

**Comment by agentschmitt** [comment]
I added a pullrequest for this, to support typescript usage of your lib.

I saw the autonumeric lib also will have typescript definitions in the next release.
https://github.com/autoNumeric/autoNumeric/pull/661

This would allow us to change options type from 'object' to ['autonumeric.options'](https://github.com/autoNumeric/autoNumeric/blob/next/index.d.ts#L321) to get even better intellisense in typescript enviroments.
